lib/EphemeralSocket.js

```diff
@@ -179,6 +179,11 @@ function allocSocket(self) {
     // When we error we should just close and de-allocate.
     function onError(err) {
         //TODO logger.error() about socket error
+
+        // First attach an error listener so consequent error
+        // events do not cause uncaught exceptions on this socket
+        self._socket.on('error', function() {});
```

Comment:
can we name this function and add a //TODO comment about logging inside the error event listeners